x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -136,32 +150,107 @@ private void triggerTasks() {
                 return;
             }
             LOGGER.info("triggering scheduled [ML] maintenance tasks");
-            executeAsyncWithOrigin(client,
-                ML_ORIGIN,
-                DeleteExpiredDataAction.INSTANCE,
-                new DeleteExpiredDataAction.Request(deleteExpiredDataRequestsPerSecond, TimeValue.timeValueHours(8)),
+            triggerDeleteExpiredDataTask();
+            triggerDeleteJobsInStateDeletingWithoutDeletionTask(
                 ActionListener.wrap(
-                    response -> {
-                        if (response.isDeleted()) {
-                            LOGGER.info("Successfully completed [ML] maintenance tasks");
-                        } else {
-                            LOGGER.info("Halting [ML] maintenance tasks before completion as elapsed time is too great");
-                        }
-                    },
-                    e -> LOGGER.error("An error occurred during maintenance tasks execution", e)));
-            auditUnassignedMlTasks(clusterService.state());
+                    response -> {},
```

Comment:
This type of action should be rare, so an info log here would be good. This way we know when ALL longer running maintenance tasks are complete since now we have two that run concurrently.

since you are using an `AcknowledgedResponse`, a predicate on `acknowledged` would work.